src/PufferVaultMainnet.sol

```diff
@@ -80,20 +129,100 @@ contract PufferVaultMainnet is PufferVault {
         }
 
         uint256 assets = previewRedeem(shares);
+
+        _updateDailyWithdrawals(assets);
+
+        _wrapETH(assets);
+
         // solhint-disable-next-line func-named-parameters
         _withdraw(_msgSender(), receiver, owner, assets, shares);
 
         return assets;
     }
 
-    // slither-disable-next-line dead-code
-    function _wrap(uint256 amount) internal {
-        _WETH.deposit{ value: amount }();
+    /**
+     * @notice Deposits native ETH
+     */
+    function depositETH(address receiver) public payable virtual returns (uint256) {
+        uint256 maxAssets = maxDeposit(receiver);
+        if (msg.value > maxAssets) {
+            revert ERC4626ExceededMaxDeposit(receiver, msg.value, maxAssets);
+        }
+
+        uint256 shares = previewDeposit(msg.value);
+        _mint(receiver, shares);
+        emit Deposit(_msgSender(), receiver, msg.value, shares);
+
+        return shares;
+    }
+
+    /**
+     * @notice Transfers ETH to a specified address
+     * @dev Restricted to PufferProtocol
+     * We use it to transfer ETH to PufferModule
+     * @param to The address to transfer ETH to
+     * @param ethAmount The amount of ETH to transfer
+     */
+    function transferETH(address to, uint256 ethAmount) external restricted {
+        // Our Vault holds ETH & WETH
+        // If we don't have enough ETH for the transfer, unwrap WETH
+        uint256 ethBalance = address(this).balance;
+        if (ethBalance < ethAmount) {
+            _WETH.withdraw(ethAmount - ethBalance);
+        }
+
+        (bool success,) = to.call{ value: ethAmount }("");
+
+        if (!success) {
+            revert ETHTransferFailed();
+        }
+    }
+
+    /**
+     * @notice Allows the `msg.sender` to burn his shares
+     * @param shares The amount of shares to burn
+     */
+    function burn(uint256 shares) public restricted {
+        _burn(msg.sender, shares);
     }
 
-    // slither-disable-next-line dead-code
-    function _unwrap(uint256 amount) internal {
-        _WETH.withdraw(amount);
+    /**
+     * @notice Sets a new daily withdrawal limit
+     * @dev Restricted to DAO
+     * @param newLimit The new daily limit to be set
+     */
+    function setDailyWithdrawalLimit(uint96 newLimit) external restricted {
+        VaultStorage storage $ = _getPufferVaultStorage();
+        emit DailyWithdrawalLimitSet($.dailyAssetsWithdrawalLimit, newLimit);
+        $.dailyAssetsWithdrawalLimit = newLimit;
+    }
+
+    function getRemainingAssetsDailyWithdrawalLimit() public view virtual returns (uint96) {
+        VaultStorage storage $ = _getPufferVaultStorage();
+        return $.dailyAssetsWithdrawalLimit - $.assetsWithdrawnToday;
```

Comment:
In the scenario that the limit is not 0 and is set to 0, then this would be negative and revert, resulting in other functions failing. 
add a check 
- $.dailyAssetsWithdrawalLimit < $.assetsWithdrawnToday return 0
- else ...